pkg/okteto/endpoint.go

```diff
@@ -0,0 +1,87 @@
+package okteto
+
+import (
+	"context"
+	"fmt"
+
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/shurcooL/graphql"
+)
+
+type endpointClient struct {
+	client graphqlClientInterface
+}
+
+func newEndpointClient(client graphqlClientInterface) *endpointClient {
+	return &endpointClient{
+		client: client,
+	}
+}
+
+type EndpointInfo struct {
+	Url graphql.String
+}
+
+type Component struct {
+	Endpoints  []EndpointInfo
+	DeployedBy graphql.String
+}
+
+type Space struct {
+	Deployments  []Component
+	Statefulsets []Component
+	Functions    []Component
+	Externals    []Component
+}
+
+type SpaceQuery struct {
+	Response Space `graphql:"space(id: $id)"`
+}
+
+func (c *endpointClient) List(ctx context.Context, ns, deployedBy string) ([]string, error) {
+	oktetoLog.Infof("listing endpoints in '%s' namespace deployed by %s", ns, deployedBy)
+	var queryStruct SpaceQuery
+	variables := map[string]interface{}{
+		"id": graphql.String(ns),
+	}
+
+	err := query(ctx, &queryStruct, variables, c.client)
+	if err != nil {
+		return nil, err
+	}
+
+	endpoints := make([]string, 0)
+	for _, deployment := range queryStruct.Response.Deployments {
+		if deployment.DeployedBy == graphql.String(deployedBy) {
```

Comment:
NIT: Given that all of them are components. Should we encapsulate into a function so we don't duplicate code?

Reply:
6e3346e02dfdfc36d5ab324fda5207cdcd1e3f96